src/content/platform-integration/ios/app-extensions.md

```diff
@@ -117,207 +212,353 @@ Choose one of the following sources for your data.
 
-You can use the following procedure or the
-[Testing on simulators](#test-on-a-simulator) instructions
-to test on physical devices.
+    *   Select **Pre-actions**.
 
-1. Launch the share extension target.
-1. In the popup window that says "Choose an app to run",
-   select an app that can be used to test share extension,
-   such as the Photos app.
-1. Select a photo, tap the share button,
-   then tap on the share extension icon of your app.
+    *   Click **+** and select **New Run Script Action**.
+
+    *   In the **Provide build settings from**
+        drop-down list, select **ShareExtension**.
+
+    *   In the **Shell** text field, enter:
+
+        ```console
+        /bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" prepare
+        ```
+
+    *   Click **Close**.
+
+1.  In Xcode, share the build configurations.
+
+    *   Open the **project navigator**.
+
+    *   In the main window under **PROJECT**, select
+        **Runner**.
+
+    *   Open the **Info** tab.
+
+    *   Expand **Configuration**.
 
-## Tutorials
+    *   Expand **Debug** and update the value for
+        **ShareExtension** to match the value for
+        **Runner**.
+
+    *   Repeat the previous step for **Profile**, and
+        **Release**.
+
+1.  (Optional) In Xcode, replace any storyboard files with
+    an extension class, if needed.
+
+    *   Open the **project navigator**.
+
+    *   Select **Runner** > **ShareExtension** > **Info**.
+
+    *   Expand **Information Property List**.
+
+    *   Delete the **NSExtensionMainStoryboard** key.
+
+    *   Add the **NSExtensionPrincipalClass** key.
+
+    *   Add one of these values for the
+        `NSExtensionPrincipalClass` key:
+
+        *   (Swift) **ShareExtension.ShareViewController**
+        *   (Objective-C)  **ShareViewController**
+
+1.  In Xcode, update the `ShareViewController` to use the
+    `FlutterViewController`.
+
+    *   Open the **project navigator**.
+
+    *   Select **Runner** > **ShareExtension** > **ShareViewController**.
+
+    *   Update `ShareViewController` to use the
+        `FlutterViewController` class:
+
+{% tabs "controller-code-tabs", true %}
+
+{% tab "UIKit-Swift" %}
+
+```swift title="ShareViewController.swift"
+import UIKit
+import Flutter
+
+class ShareViewController: UIViewController {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        showFlutter()
+    }
+
+    func showFlutter() {
+        let flutterViewController = FlutterViewController(project: nil, nibName: nil, bundle: nil)
+        addChild(flutterViewController)
+        view.addSubview(flutterViewController.view)
+        flutterViewController.view.frame = view.bounds
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        self.extensionContext?.cancelRequest(
+            withError: NSError(domain: Bundle.main.bundleIdentifier!, code: 0))
+    }
+}
+```
+
+{% endtab %}
+
+{% tab "UIKit-ObjC" %}
```

Comment:
@hellohuanlin I need help with two things:

* I added a section that shows how to update the `.h` file and the `.m` file for Objective-C. Can you double-check to make sure that what I did is correct?

* I remove the registrant code from Objective-C. Did I remove everything correctly?

Reply:
it's correct (to both questions)

Reply:
^_^